### ISSUE TITLE
ci: Add the ability to bring up the application using a production build

### ISSUE THREAD

**Issue by johncowen** [kumahq/kuma-gui#819]
### Description

TLDR; In the interest of trying to make smaller issues, this covers bringing up a production build in CI for things like Cypress to test against.

### The Task:

- Build a production build and serve somewhere/somehow that CI can access it.
- Ensure that CI can also tear this down once the tests have finished.

### Background info:

We are currently working on https://github.com/kumahq/kuma-gui/pull/803 which begins the infra for running frontend e2e tests in a browser runner, in our case Cypress. Cypress runs in a separate process to the application itself (different to when running the application during development or during jest testing) and requires a separate running application to perform tests against.

We have existing work which uses the PR preview builds to run against, but as the preview builds use MSW to do our data mocking, this means we cannot easily massage data for our test setup (arranging) easily as we are not in the same process as MSW.

Cypress (and the work in https://github.com/kumahq/kuma-gui/pull/803) uses `cy.intercept` instead to intercept network requests from within the 'Cypress process' for network requests made from the 'application process'. This means we can (and should) build a full production "this is the thing we ship" build and run that to test against.

As the work in https://github.com/kumahq/kuma-gui/pull/803 intercepts/mocks every single network request as well as adds the ability to massage individual ones, this full production build doesn't need an actual production backend to provide a HTTP API.

Therefore the work here should involve a `yarn build` to produce our production artifact, then provide some sort of server in CI to to serve that artifact. This build should be kept running for the duration of the Cypress tests and then automatically torn down once the tests are complete.

One final note that I realized after our conversation here is that I don't think we can use some sort of off-the-shelf node server like Vercel's `serve` to serve the artifact, as we will need to add the ability to do our index.html file variable interpolation on the production index.html file.

Alternatively we could do a build that is "this is a full production build, apart from the index.html" file i.e. we could run `yarn build` and then interpolate the index.html file statically, rather than dynamically interpolating the the html file at runtime.

Dynamically interpolating would be my preference as its closest to what we do in prod, but if its troublesome for whatever reason then the static interpolation would be a good second best. Final thought that came to mind whilst writing this (:rubber-duck:), I guess if we use static post build interpolation we could use the same method we use for our preview sites to run the thing? Just it would only work when using Cypress due to Cypress mocking out the backend via `cy.intercept` 🤔 , this might even make me prefer the static approach.

Once this task/issue is complete, we are in a place where we can integrate the work in https://github.com/kumahq/kuma-gui/pull/803 into CI, for which I'll make a separate issue.

**Comment by johncowen** [comment]
Closed by https://github.com/kumahq/kuma-gui/pull/829
